api/src/opentrons/protocol_api/deck.py

```diff
@@ -0,0 +1,136 @@
+"""Deck state accessors for the Protocol API."""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Iterator, List, Mapping, Optional, Tuple, Union
+
+from opentrons_shared_data.deck.dev_types import SlotDefV3
+from opentrons.types import DeckLocation, Location, Point
+
+from .core.common import ProtocolCore
+from .labware import Labware
+from .module_contexts import ModuleContext
+
+DeckItem = Union[Labware, ModuleContext]
+
+
+@dataclass(frozen=True)
+class CalibrationPosition:
+    """A calibration point on the deck of the robot.
+
+    Attributes:
+        id: Unique identifier for the calibration point.
+        position: The absolute x, y, z coordinate of the point.
+        displayName: A human-readable nickname for this point.
+    """
+
+    id: str
+    position: Tuple[float, float, float]
+    displayName: str
+
+
+class AbstractDeck(ABC, Mapping[DeckLocation, DeckItem]):
+    """Abstract interface for :py:class:`Deck`."""
+
+    @abstractmethod
+    def __getitem__(self, key: DeckLocation) -> DeckItem:
```

Comment:
Probably try to preserve `deck["3"] is None` if slot is empty. A bit funny for a "dict-like" interface, but breaking the existing API here seems unnecessary

```suggestion
class AbstractDeck(ABC, Mapping[DeckLocation, Optional[DeckItem]]):
    """Abstract interface for :py:class:`Deck`."""

    @abstractmethod
    def __getitem__(self, key: DeckLocation) -> Optional[DeckItem]:
```